src/charm.py

```diff
@@ -149,25 +159,28 @@ def _get_interfaces(self):
             raise ErrorWithStatus(err, BlockedStatus)
         return interfaces
 
-    def _get_relational_db_data(
-        self,
-    ):
-        relational_db = self.model.relations["relational-db"]
-        if len(relational_db) > 1:
-            raise ErrorWithStatus(
-                f"Too many mysql relations. Found {len(relational_db)}, expected 1", BlockedStatus
-            )
-
-        try:
-            relational_db = relational_db[0]
-            db_unit = list(relational_db.units)[0]
-            relational_db = relational_db.data[db_unit]
-            relational_db["database"]
-            return relational_db
-        except IndexError:
-            raise ErrorWithStatus("Waiting for relational-db relation data", WaitingStatus)
-        except KeyError:
-            raise ErrorWithStatus("Missing data in relational-db relation", WaitingStatus)
+    def _get_relational_db_data(self) -> dict:
+        mysql_relation = self.model.get_relation("relational-db")
```

Comment:
What can we check here except for `None` to ensure the relation is established?
The flow here is to check relation to make sure this charm is related to MySQL charm, but retrieve data through library.

Reply:
get_relation only returns if relation is established ... othervise its none for the relation